sdk/daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/crypto/Hash.scala

```diff
@@ -299,11 +550,17 @@ object Hash {
       }
   }
 
-  private final class UpgradeFriendlyBuilder(purpose: Purpose, cid2Bytes: Value.ContractId => Bytes)
-      extends ValueHashBuilder(
+  private final class UpgradeFriendlyBuilder(
```

Comment:
add a comment here why it is "upgrade friendly"?

Reply:
There's already a big comment below explaining the reasoning

Reply:
ah ok, that didn't show up in the PR diff. does it make sense to move the comment to the class?